tests/datadoc/test_dataset.py

```diff
@@ -538,17 +538,78 @@ def test_datadoc():
         SEMDATA["SEM_cement_batch2/77600-23-001/77600-23-001_5kV_400x_m001"],
     }
 
+    # Filter on criterion, but without required value
+    assert set(
+        search(
+            ts,
+            criteria={"creator.name": ""},
```

Comment:
I think that the only None should be treated as a wildcard. The empty string should only match an empty string.